ui/src/vertical-geometry/coordinates.ts

```diff
@@ -0,0 +1,30 @@
+export interface Coordinates {
+    startM: number;
+    endM: number;
+    meterHeightPx: number;
+    mMeterLengthPxOverM: number;
+    fullDiagramHeightPx: number;
+    diagramWidthPx: number;
+    horizontalTickLengthMeters: number;
+
+    bottomHeightPaddingPx: number;
+    bottomHeightTick: number;
+    topHeightTick: number;
+    chartHeightPx: number;
+}
+
+export function mToX(coordinates: Coordinates, m: number): number {
+    return (m - coordinates.startM) * coordinates.mMeterLengthPxOverM;
+}
+
+export function xToM(coordinates: Coordinates, x: number): number {
+    return x / coordinates.mMeterLengthPxOverM + coordinates.startM;
```

Comment:
Jos (ja oletettavasti kun) tämän ja tuon ylemmän funktion pitäisi olla toistensa komplementit, niin tässäkin oletettavasti pitäisi olla sulut tuon pluslaskun ympärillä, että laskujärjestys menee oikein

Reply:
Nuo käänteisfunktiot hämmensi minuakin, muuuuttta kyllähän nuo oikeinpäin menee juuri näin, kun komposition käänteisfunktiohan menee niin että (f∘g)<sup>-1</sup> = g<sup>-1</sup>∘f<sup>-1</sup>.

Tai Haskell-syntaksilla (kompositio luetaan oikealta vasemmalle) juuri tätä tapausta katsoen, ja vähän muuttujien nimiä lyhentäen, kun on `mToX = (* mMeterLength) . (subtract startM)`; niin sen käänteisfunktio on `xToM = (+ startM) . (/ mMeterLength)`